From 009fd527b0f49aaae088d28a731930b5544c647a Mon Sep 17 00:00:00 2001
From: Sai Kumar Sanagavarapu <ssanagav@codeaurora.org>
Date: Tue, 5 Jul 2016 12:03:38 +0530
Subject: [PATCH 2/9] CameraService: Fix deadlock in binder death cleanup.

Issue:
In the event of a binder death, there is a chance of deadlock
due to recursive lock acquisition in the death handling sequence.

Fix:
Clear evicted client list before acquiring service lock.

CRs-Fixed: 984249
Change-Id: I6fc5fa6e01c002bc46be058fcd977be14cae0270
---
 services/camera/libcameraservice/CameraService.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/CameraService.cpp b/services/camera/libcameraservice/CameraService.cpp
index 3755d55cd..576663d6a 100644
--- a/services/camera/libcameraservice/CameraService.cpp
+++ b/services/camera/libcameraservice/CameraService.cpp
@@ -1787,7 +1787,8 @@ bool CameraService::evictClientIdByRemote(const wp<IBinder>& remote) {
                 ret = true;
             }
         }
-
+        //clear the evicted client list before acquring service lock again.
+        evicted.clear();
         // Reacquire mServiceLock
         mServiceLock.lock();
 
-- 
2.15.1

